File: adobe-edge/android/src/main/java/com/theoplayer/reactnative/adobe/edge/AdobeEdgeConnector.kt
package com.theoplayer.reactnative.adobe.edge

import com.adobe.marketing.mobile.LoggingMode
import com.adobe.marketing.mobile.edge.identity.IdentityMap
import com.theoplayer.android.api.player.Player

class AdobeEdgeConnector(
  player: Player,
  trackerConfig: Map<String, String>,
  customIdentityMap: IdentityMap? = null,
) {

  private val handler = AdobeEdgeHandler(player, trackerConfig, customIdentityMap)

  fun updateMetadata(metadata: HashMap<String, String>) {
    handler.updateMetadata(metadata)
  }

  fun setCustomIdentityMap(customIdentityMap: IdentityMap) {
    handler.setCustomIdentityMap(customIdentityMap)
  }

  fun stopAndStartNewSession(metadata: Map<String, String>?) {
    handler.stopAndStartNewSession(metadata)
  }

  fun setLoggingMode(loggingMode: LoggingMode) {
    handler.setLoggingMode(loggingMode)
  }

  fun setError(errorId: String) {
    handler.setError(errorId)
  }

  fun destroy() {
    handler.destroy()
  }
}

File: adobe-edge/ios/Connector/AdobeEdgeConnector.swift
//
//  AdobeEdgeConnector.swift
//

import Foundation
import THEOplayerSDK
import UIKit
import AEPServices
import AEPEdgeIdentity

class AdobeEdgeConnector {
    private var handler: AdobeEdgeHandler
    init(player: THEOplayer, trackerConfig: [String:String], customIdentityMap: [String:Any]? = nil) {
        self.handler = AdobeEdgeHandler(player: player, trackerConfig: trackerConfig, customIdentityMap: customIdentityMap)
    }
    
    func updateMetadata(_ metadata: [String:String]) -> Void {
        self.handler.updateMetadata(metadata)
    }
    
    func setCustomIdentityMap(_ customIdentityMap: [String:Any]) -> Void {
        self.handler.setCustomIdentityMap(customIdentityMap)
    }
    
    func stopAndStartNewSession(_ metadata: [String:String]) -> Void {
        self.handler.stopAndStartNewSession(metadata)
    }
    
    func setLoggingMode(_ debug: LogLevel) -> Void {
        self.handler.setLoggingMode(debug)
    }
    
    func setError(_ errorId: String) -> Void {
        self.handler.setError(errorId)
    }
    
    func destroy() -> Void {
        self.handler.destroy()
    }
}
